diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 7c81c3102f..2ee4a69873 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -7317,20 +7317,6 @@ static bool setting_append_list(
                &subgroup_info,
                parent_group);
 
-#if !defined(IOS) && !defined(HAVE_LAKKA)
-         if (frontend_driver_has_fork())
-         {
-            CONFIG_ACTION(
-                  list, list_info,
-                  MENU_ENUM_LABEL_RESTART_RETROARCH,
-                  MENU_ENUM_LABEL_VALUE_RESTART_RETROARCH,
-                  &group_info,
-                  &subgroup_info,
-                  parent_group);
-            menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_RESTART_RETROARCH);
-         }
-#endif
-
          CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_CONFIGURATIONS_LIST,
@@ -7427,17 +7413,6 @@ static bool setting_append_list(
                parent_group);
          menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_UI_COMPANION_TOGGLE);
 #endif
-#if !defined(IOS)
-         /* Apple rejects iOS apps that let you forcibly quit them. */
-         CONFIG_ACTION(
-               list, list_info,
-               MENU_ENUM_LABEL_QUIT_RETROARCH,
-               MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
-               &group_info,
-               &subgroup_info,
-               parent_group);
-         menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_QUIT);
-#endif
 
 #if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
         CONFIG_ACTION(
@@ -7449,7 +7424,6 @@ static bool setting_append_list(
               parent_group);
 #endif
 
-#if defined(HAVE_LAKKA)
 #ifdef HAVE_LAKKA_SWITCH
         CONFIG_ACTION(
                list, list_info,
@@ -7484,7 +7458,6 @@ static bool setting_append_list(
                &subgroup_info,
                parent_group);
          menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_SHUTDOWN);
-#endif
 
          CONFIG_ACTION(
                list, list_info,
@@ -12812,24 +12785,6 @@ static bool setting_append_list(
                   general_write_handler,
                   general_read_handler,
                   SD_FLAG_NONE);
-#else
-#if !defined(IOS)
-            if (frontend_driver_has_fork())
-               CONFIG_BOOL(
-                     list, list_info,
-                     &settings->bools.menu_show_restart_retroarch,
-                     MENU_ENUM_LABEL_MENU_SHOW_RESTART_RETROARCH,
-                     MENU_ENUM_LABEL_VALUE_MENU_SHOW_RESTART_RETROARCH,
-                     menu_show_restart_retroarch,
-                     MENU_ENUM_LABEL_VALUE_OFF,
-                     MENU_ENUM_LABEL_VALUE_ON,
-                     &group_info,
-                     &subgroup_info,
-                     parent_group,
-                     general_write_handler,
-                     general_read_handler,
-                     SD_FLAG_NONE);
-#endif
 #endif
 
 #if defined(HAVE_XMB) || defined(HAVE_OZONE)
